addons/im_livechat/static/src/core/common/transcript_sender.js

```diff
@@ -23,23 +21,47 @@ export class TranscriptSender extends Component {
 
     setup() {
         this.isValidEmail = isValidEmail;
-        this.livechatService = useService("im_livechat.livechat");
+        useAutofocus();
+        this.store = useService("mail.store");
         this.state = useState({
-            email: "",
+            email: this.initialEmail,
             status: this.STATUS.IDLE,
         });
     }
 
+    get initialEmail() {
+        return "";
+    }
+
+    get isButtonDisabled() {
+        return (
+            !this.state.email ||
+            !this.isValidEmail(this.state.email) ||
+            [this.STATUS.SENDING, this.STATUS.SENT].includes(this.state.status)
+        );
+    }
+
+    /** @param {KeyboardEvent} ev */
+    onKeydown(ev) {
+        if (ev.key == "Enter" && !this.isButtonDisabled) {
+            this.onClickSend();
+        }
+    }
+
     async onClickSend() {
         this.state.status = this.STATUS.SENDING;
         try {
             await rpc("/im_livechat/email_livechat_transcript", {
                 channel_id: this.props.thread.id,
                 email: this.state.email,
             });
-            this.state.status = this.STATUS.SENT;
+            this.onTranscriptSent();
         } catch {
             this.state.status = this.STATUS.FAILED;
         }
     }
+
+    onTranscriptSent() {
+        this.state.status = this.STATUS.SENT;
+    }
```

Comment:
This method can be removed if we just provide the `onSend` prop and call it after calling the rpc.